File: src/main/java/org/apache/prepbuddy/cluster/ClusteringAlgorithm.java
package org.apache.prepbuddy.cluster;

import scala.Tuple2;

import java.util.List;

/**
 * ClusteringAlgorithm is for implementing the algorithm which can be use to
 * clustering the column value
 */
public interface ClusteringAlgorithm {
    Clusters getClusters(List<Tuple2<String, Integer>> tuples);
}

File: src/main/scala/org/apache/datacommons/prepbuddy/clusterers/ClusteringAlgorithm.scala
package org.apache.datacommons.prepbuddy.clusterers

/**
  * ClusteringAlgorithm is for implementing the algorithm which can be use to clustering the column value
  */
trait ClusteringAlgorithm extends Serializable {
    def getClusters(tuples: Array[(String, Int)]): Clusters
}
